### PR TITLE
Make the EventsTable poll less

### PR DESCRIPTION
**Pull request by pauldambra**
## Changes

Includes two changes

1) always give the `eventsTableLogic` a key to avoid having two copies running and both polling
2) listens to the page visibility change event and pause polling when the page is hidden

### logics polling before

Screenshot 2021-12-31 at 23 32 42

### logics polling after

Screenshot 2021-12-31 at 22 32 23

### new behaviour

events-polling

## How did you test this code?

Adding unit tests for changes.

Also running the site locally and opening developer tools network view in a separate window. Watch that the events table is polling, and then minimise the browser or switch tabs and see that the event table stops polling

### CODE REVIEW

**Review by Twixes — CHANGES_REQUESTED** [review]
Good improvement, but behavior is wonky with `afterMount`!

File: frontend/src/scenes/events/EventsTable.tsx [hunk 1]
```diff
@@ -74,11 +74,38 @@ export function EventsTable({
...
+        const onVisibilityChange = (): void => {
+            setPollingActive(!document[hidden])
+        }
+
+        document.addEventListener(visibilityChange, onVisibilityChange)
+
+        return function cleanUp() {
+            document.removeEventListener(visibilityChange, onVisibilityChange)
+        }
+    }, [])
```
[the remark this answers is not in the document]
Comment (neilkakkar) [reply frontend/src/scenes/events/EventsTable.tsx:107]: Not sure if similar enough, but I built a visibilitySensor a while back for something else: https://github.com/PostHog/posthog/blob/master/frontend/src/lib/components/VisibilitySensor/VisibilitySensor.tsx#L1

Perhaps you can reuse it here?

[the remark this answers is not in the document]
Comment (pauldambra) [reply frontend/src/scenes/events/EventsTable.tsx]: If I'm reading that correctly the `VisibilitySensor` is whether a given element is within the viewport. Whereas the `usePageVisibility` is whether the browser tab/page is currently visible

[the remark this answers is not in the document]
Comment (neilkakkar) [reply frontend/src/scenes/events/EventsTable.tsx]: Ah, my bad! Makes sense

**Review by Twixes — APPROVED** [review]
Works perfect now! Only issue is something (erm #7855) merged a few minutes ago caused merge conflicts…

**Comment by pauldambra** [comment]
@Twixes couple of small changes on confirming that #7855 and this work together

@macobo FYI we've overlapped - I've checked that the Actions page doesn't poll and my pausing still work now merged together. 

It's tempting to refactor the naming and overlap between `disableActions` and `setPollingActive`. But I'd suggest we do that as a follow-up (if at all)

after-merge

**Review by Twixes — APPROVED** [review]
All good after merge

**Pull request merged**